client/packages/system/src/Item/context/useUnitVariant.tsx

```diff
@@ -1,71 +1,79 @@
+import { create } from 'zustand';
+import isEqual from 'lodash/isEqual';
+import { UnitVariantNode } from '@common/types';
 import { NumUtils } from '@common/utils';
+import { useEffect } from 'react';
 
-export const useUnitVariant = (itemId?: string) => {
-  const { data } = useItemVariants();
-  const item = data?.find(item => item.id === itemId);
-
-  const asPackUnit = (packSize: number) => {
-    return (
-      item?.variants.find(variant => variant.packSize === packSize)
-        ?.shortName || `${item?.unit} x ${packSize}`
-    );
-  };
-
-  // calculates SOH, AMC, MOS, Target, Suggested and Requested based on default or user selected variant
-  const numberOfPacksFromQuantity = (
-    totalQuantity: number,
-    packSize: number
-  ) => {
-    return NumUtils.round(totalQuantity / packSize, 2);
-  };
-
-  const options = item?.variants ?? [];
-
-  // const quantityFromNumberOfPacks - reverse of above
-
-  // set in app data
-  const setDefaultOption = () => {};
-
-  const defaultOption = () => {
-    // TODO: search app data first to see if user has preferred preference
-    return item?.variants.find(variant =>
-      variant.longName.includes(item?.unit)
-    );
+interface UnitState {
+  // From back end
+  items: {
+    [itemId: string]: UnitVariantNode;
   };
+  // Should be called on startup when fetching multi unit variants
+  setItems: (newItems: UnitVariantNode[]) => void;
+}
 
+const useUnitStore = create<UnitState>(set => {
   return {
-    asPackUnit,
-    numberOfPacksFromQuantity,
-    options,
-    setDefaultOption,
-    defaultOption,
+    items: {},
+    userSelected: {},
+    // TODO add user selected
+
+    setItems: newItems =>
+      set(() => {
+        const items = newItems.reduce(
+          (acc, item) => ({ [item.itemId]: item, ...acc }),
+          {}
+        );
+        return { items };
+      }),
   };
+});
+
+export const useInitUnitStore = () => {
+  const { setItems } = useUnitStore();
+  // This should happen on startup and when store is changed (for store changed, calculated mostUsed and app data userSelecte would change)
+  // Suggested places:
+  // https://github.com/openmsupply/open-msupply/blob/312b837c3d17a1ead05e140b7668cd5f45dffbc3/client/packages/common/src/authentication/api/hooks/useLogin.ts#L107
+  // https://github.com/openmsupply/open-msupply/blob/312b837c3d17a1ead05e140b7668cd5f45dffbc3/client/packages/common/src/authentication/AuthContext.tsx#L125
+  const { data } = useUnitVariantList();
+
+  useEffect(() => {
+    if (!data) return;
+    setItems(data || []);
+  }, [data]);
+
+  // TODO add user selected from app data
 };
 
-export type UseUnitVariantContext = ReturnType<typeof useUnitVariant>;
-
-const UnitVariantContext = createContext<UseUnitVariantContext>({} as any);
+export const useUnitVariant = (itemId: string) => {
+  // TODO [state.items[itemId], state.userSelected[itemdId]]
+  const item = useUnitStore(state => state.items[itemId], isEqual);
+  if (!item) {
```

Comment:
I like this early return, rather then constant use of `?`, even though it may end up being more code